snapcraft.yaml

```diff
@@ -0,0 +1,113 @@
+name: desmume-emulator
+version: 0.9.13
+summary: DeSmuME is a Nintendo DS Emulator
+description: |
+  An open-source Nintendo DS Emulators.
+adopt-info: desmume-emulator
+grade: stable
+confinement: strict
+base: core22
+compression: lzo
+parts:
+  desmume-emulator:
+    source: https://github.com/TASEmulators/desmume/archive/refs/tags/release_0_9_13.tar.gz
+    plugin: nil
+    override-build: |
+      LDFLAGS="" CFLAGS="" FFLAGS="" FCFLAGS="" DEB_CFLAGS_SET="" DEB_CXXFLAGS_SET="" DEB_FFLAGS_SET="" DEB_FCFLAGS_SET="" DEB_LDFLAGS_SET="" meson setup --prefix=${CRAFT_PART_INSTALL}/usr desmume/src/frontend/posix/build desmume/src/frontend/posix
+      LDFLAGS="" CFLAGS="" FFLAGS="" FCFLAGS="" DEB_CFLAGS_SET="" DEB_CXXFLAGS_SET="" DEB_FFLAGS_SET="" DEB_FCFLAGS_SET="" DEB_LDFLAGS_SET="" ninja -C desmume/src/frontend/posix/build
+      LDFLAGS="" CFLAGS="" FFLAGS="" FCFLAGS="" DEB_CFLAGS_SET="" DEB_CXXFLAGS_SET="" DEB_FFLAGS_SET="" DEB_FCFLAGS_SET="" DEB_LDFLAGS_SET="" ninja install -C desmume/src/frontend/posix/build
+
+    stage-packages:
+      - libasound2 
+      - libc6 
+      - libcairo2 
+      - libgcc-s1
+      - libgcc-s1      
+      - libgdk-pixbuf-2.0-0 
+      - libgl1
+      - libglade2-0 
+      - libglib2.0-0 
+      - libglu1-mesa
+      - libgtk2.0-0 
```

Comment:
you dont need to install gtk2-related packages if you use the meson recipe, it builds only gtk+3 support